tls/s2n_client_hello.h

```diff
@@ -43,6 +43,14 @@ struct s2n_client_hello {
      * issues a hello retry.
      */
     unsigned int parsed : 1;
+
+    /*
+     * SSLv2 ClientHellos have a different format.
+     * Cipher suites are each three bytes instead of two.
+     * And due to how s2n-tls parses the record,
+     * the raw_message will not contain the protocol version.
+     */
+    unsigned int sslv2 : 1;
```

Comment:
Not sure if this would really be any better, but if you put the client hello version in this struct you could check the version for sslv2 and also not have to parse it again.

Reply:
I was thinking I couldn't move client_hello_version from s2n_connection to s2n_client_hello because s2n_connection_free_handshake frees the client_hello, but it turns out[ it only free the raw_message](https://github.com/aws/s2n-tls/blob/f2faa0e25b1d68cd36173ae44df58be3218b6ca1/tls/s2n_connection.c#L416-L422). Which seems like a recipe for memory errors since the blobs in s2n_client_hello point into raw_message, but that's a separate problem.

I'll move client_hello_version (and maybe clean up s2n_connection_free_handshake a bit).

Reply:
Moved client_hello_version: https://github.com/aws/s2n-tls/pull/3819 (it ended up touching a lot, so made a new PR)

Opened an issue to fix s2n_connection_free_handshake: https://github.com/aws/s2n-tls/issues/3820